client/src/app/components/page/list/list.component.ts

```diff
@@ -1,15 +1,24 @@
 import { Component, OnInit } from '@angular/core';
-
+import {TaskService} from '../../../services/task.service';
+import {Task} from '../../../models/task';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 @Component({
   selector: 'app-list',
   templateUrl: './list.component.html',
   styleUrls: ['./list.component.scss']
 })
 export class ListComponent implements OnInit {
-
-  constructor() { }
+  public tasks:Task[];
+  private ngUnsubscribe = new Subject<void>();
+  constructor(private taskService: TaskService) { }
 
   ngOnInit() {
+    this.getAllTasks();
+  }
+  public getAllTasks(){
+    this.taskService.getAllTasks().pipe(takeUntil(this.ngUnsubscribe)).subscribe(tasks=>{
```

Comment:
please implement OnDestroy method in all paces where you have subscriptions
and complete your this.ngUnsubscribe there

Reply:
I added
```
    ngOnDestroy() {
        this.ngUnsubscribe.next()
        this.ngUnsubscribe.complete()
    }
```

Reply:
No, you added onDestroy, but need to be ngOnDestroy *

Reply:
My bad, changed it.